### PR TITLE
Add hints to use macro packages for new users

### PR DESCRIPTION
**Pull request by bkamins**
Users asked to give a hint that macro-packages can be easier to use for newcomers in the opening page of DataFrames.jl manual near its top (so that it is easily discoverable).

@nalimilan, @pdeffebach, @jkrumbiegel - I have proposed something, but I am not sure if the wording is best. Can you please comment? (I think that good wording here is important) Thank you!

Also I mention only DataFramesMeta.jl as I did not want to introduce additional user confusion (i.e. that they need to make choices between DataFramesMeta.jl and DataFrameMacros.jl). But maybe it is better to mention both. I am open for suggestions.
(they are both mentioned below in data wrangling section)

### CODE REVIEW

**Comment by pdeffebach** [comment]
I think this looks good, thanks!

**Comment by jkrumbiegel** [comment]
One the one hand I'm ok with the text as it is, on the other you might add something like "or one of the other convenience packages" or so and just intra-link to the section where they are shown. Often people just stick with what they encounter first, so I personally like laying out the options early. Unless you want to endorse DataFramesMeta as the macro-arm of DataFrames, which you can of course do and might be a good decision, too, from an ecosystem building perspective. I'm happy if people find something that works for them in the Julia ecosystem, my only worry regarding DataFrameMacros was always that it would mostly be disregarded because it seems not to be the "official" package (without understanding anything about the differences), not because it's not good.

**Comment by jkrumbiegel** [comment]
Sounds good, thank you!

File: docs/src/index.md [hunk 1]
```diff
@@ -34,8 +34,15 @@ tool for working with tabular data in Julia -- as noted below, there are some
...
-users can start with the [First Steps with DataFrames.jl](@ref) section.
+To understand the toolchain in more detail, have a look at the tutorials in this manual.
+New users can start with the [First Steps with DataFrames.jl](@ref) section.
+
+If you do not have a significant programming experience you might find 
+[DataFramesMeta.jl](https://juliadata.github.io/DataFramesMeta.jl/stable/) package
+or one of the other convenience packages discussed in
+[Data manipulation frameworks](@ref) section of this manual
+helpful when writing more advanced data transformations. These packages provide
+convenience syntax similar to [dplyr](https://dplyr.tidyverse.org/) in R.
```
Comment (nalimilan) [inline docs/src/index.md:45]: Maybe mention programming experience at the end, as these macros are convenient even if you're an experienced programmer?
```suggestion
You may find the [DataFramesMeta.jl](https://juliadata.github.io/DataFramesMeta.jl/stable/)
package or one of the other convenience packages discussed in
the [Data manipulation frameworks](@ref) section of this manual
helpful when writing more advanced data transformations,
especially if you do not have a significant programming experience.
These packages provide convenience syntax similar to [dplyr](https://dplyr.tidyverse.org/) in R.
```

**Review by nalimilan — APPROVED** [review]

**Comment by bkamins** [comment]
Thank you!

**Pull request merged**